### ISSUE TITLE
No connection to referee remote control

### ISSUE THREAD

**Issue by g3force** [robotics-erlangen/autoref#2]
I was not able to connect the autoref to the ssl-referee remote control (port 10007) yet.

I enabled the remote control in ssl-refbox and made sure that remote control is enabled in autoref on same port (10007). But no connection is established to ssl-refbox.

I also successfully supplied vision data using the ssl-tools logplayer, but this had no effect on the remote control.

Any suggestions?

**Comment by MichaelEischer** [comment]
The autoref should connect to the refbox when issuing its first command. The refbox is expected to run locally, that is the hard-coded remote control address is localhost:10007 and have remote control enabled in the config menu.
If you don't have some vision data that should cause the autoref to issue commands, you can apply the following patch to `autoref/init.lua`:
```lua
 local function mainLoopWrapper(func)
     return function()
+        Refbox.send("STOP")
         if not World.update() then
             return -- skip processing if no vision data is available yet
```

**Comment by g3force** [comment]
when applying the patch, the autoref connects to refbox, thx.

But I still can not get the autoref to start sending commands without the patch.

My setup is:
1. ssl-refbox running with remote control enabled
2. graphical client to verify vision
3. Tigers Simulator with artificial vision
4. autoref

Issues:
1. I can not see the ball in autoref, but it is visible in graphical client
2. autoref can not parse Geometry (I get the same errors when using ssl-vision):
```
[libprotobuf ERROR google/protobuf/message_lite.cc:121] Can't parse message of type "SSL_WrapperPacket" because it is missing required fields: geometry.field.goal_width, geometry.field.goal_depth, geometry.field.goal_wall_width, geometry.field.center_circle_radius, geometry.field.defense_radius, geometry.field.defense_stretch, geometry.field.free_kick_from_defense_dist, geometry.field.penalty_spot_from_field_line_dist, geometry.field.penalty_line_from_spot_dist
```
3. Putting bots in their own penArea or letting the ball leaving the field does not trigger a referee command 

Setup looks like this:
![image](https://user-images.githubusercontent.com/779094/33684929-3673a562-dad0-11e7-9727-796089a8b8e3.png)

**Comment by MichaelEischer** [comment]
The root cause of the problems is probably the inability to parse the geometry (Issue 2). Currently our autoref only supports the legacy field geometry format, thus the parsing error. However, without geometry which also includes camera positions, the ball tracking doesn't work (Issue 1). This leaves you with just the robots. As playing a game without a ball makes no sense, the autoref requires that the ball has been visible at least once, otherwise it refuses to work (Issue 3).
We plan to implement the new geometry format in about one or two months. Fixing the tracking to work without camera calibration data also won't happen before the new geometry format is implemented.
I'm not sure whether there is a viable short term fix. You could comment out the code that check whether there is a ball https://github.com/robotics-erlangen/autoref/blob/master/autoref/init.lua#L85 , but that may cause further issues.

**Comment by g3force** [comment]
Ok, I see...

I also tested it with ssl-vision in replay mode. Legacy geometry publishing is activated. I still get the parsing error :/

Shouldn't it work with the logplayer? The logrecorder records referee, vision and legacy vision, so there should be the legacy geometry message as well. But I also get the parsing error with the logplayer :/

Running the autoRef without a ball sounds quite useless ;)

**Comment by MichaelEischer** [comment]
Did you select the legacy vision port?

**Comment by g3force** [comment]
Good point, I thought it listens to the default legacy port automatically...

Now I receive the ball and there are no parse errors :)

And I also receive referee commands.
